src/Service.php

```diff
@@ -496,8 +496,17 @@ public function doDeleteFromIndex(Concrete $object): void
             'id' => $object->getId()
         ];
 
-        $this->logger->info('Deleting data object ' . $object->getId() . ' from es index.');
-        $this->getClient()->delete($params);
+        try {
+            $exists = $this->getClient()->get($params);
+        } catch (Exception $e){
+            $exists = false;
+            $this->logger->debug($e->getMessage());
+            return;
```

Comment:
There's no need to set `$exist` to `false` if you `return` right after.

```suggestion
            $this->logger->debug($e->getMessage());
            return;
```

Reply:
My bad, this return was left there from a different approach I was working on where I would return immediately on a caught exception.
I have removed it, as I prefer to validate for the return value of the get function just in case in the future opensearch project updates their client to return false on indexes not found rather than throwing an exception. It is more future proof. But either approach works.